src/cfnlint/schemaManager.py

```diff
@@ -0,0 +1,103 @@
+"""
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+
+import json
+import logging
+import os
+import platform
+
+import botocore.exceptions
+import boto3
+from cfnlint.template import Template
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SchemaManager(object):
+    """Download schemas if necessary and validate modules"""
+    def __init__(
+            self, filename, template, regions):
+        self.filename = filename
+        self.regions = regions
+        self.template = Template(filename, template, regions)
+        self.boto3_sts = boto3.client('sts')
+        self.boto3_cfn = boto3.client('cloudformation')
+
+    # Check if the appropriate folder already exists
+    def check_folders(self, name, registry_type):
+        account_id = self.boto3_sts.get_caller_identity().get('Account')
+        username = None
+        path_split = os.getcwd().split('/')
```

Comment:
We should talk through what the goal is here.  Are we trying to put the files where cfn-lint is installed? into a user cache folder?  Doesn't cwd return where we are running cfn-lint?  What if it isn't in the users folder?

Reply:
Additionally what if it its a docker container? So many cfn-lint runs are inside a docker container.